Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -541,16 +545,26 @@ else if (shouldSetSelection) {
 
 
   @NotNull
-  private static CompletionSuggestion createCompletionSuggestionFromAvailableSuggestion(@NotNull final AvailableSuggestion suggestion,
-                                                                                        @NotNull final AvailableSuggestionSet suggestionSet,
-                                                                                        final int relevance) {
+  private static CompletionSuggestion createCompletionSuggestionFromAvailableSuggestion(
+      @NotNull final AvailableSuggestion suggestion,
+      @NotNull final AvailableSuggestionSet suggestionSet,
+      final int suggestionSetRelevance,
+      Map<String, IncludedSuggestionRelevanceTag> includedSuggestionRelevanceTags) {
+    int relevance = suggestionSetRelevance;
+    for (String tag : suggestion.getRelevanceTags()) {
+      if (!includedSuggestionRelevanceTags.containsKey(tag)) {
+        continue;
+      }
+
+      relevance += includedSuggestionRelevanceTags.get(tag).getRelevanceBoost();
```

Comment:
FYI - I was investigating an issue with relevance/sorting in VS Code (https://github.com/Dart-Code/Dart-Code/issues/1742) and noticed this in the comments for tags:

```
         * If an AvailableSuggestion has relevance tags that match more
	 * than one IncludedSuggestionRelevanceTag, the maximum
	 * relevance boost is used.
	 */
```

My code is currently adding them together (which I think I borrowed from here :-)) but it seems like that might be incorrect. I'm not sure who is best to confirm (@scheglov?).

Reply:
I don't remember if we made a decision to do otherwise, so probably we should follow the documentation recommendation.

Reply:
Thanks for reminding me about this @DanTup! We went back and forth on this a bit but I think we should use the maximum. I will go ahead and make a CL for the IntelliJ plugin to reflect that.